CommonMark/inlines.py

```diff
@@ -62,7 +67,7 @@
 reLinkLabel = re.compile('^\\[(?:[^\\\\\\[\\]]|' + ESCAPED_CHAR +
                          '|\\\\){0,1000}\\]')
 # Matches a string of non-special characters.
-reMain = re.compile(r'^[^\n`\[\]\\!<&*_\'"]+', re.MULTILINE)
+reMain = re.compile(r'^[^\n`\[\]\\!<&*_\'"~\^]+', re.MULTILINE)
```

Comment:
This regex is copying the behavior of the corresponding one in commonmark.js: https://github.com/commonmark/commonmark.js/blob/master/lib/inlines.js#L85

So we don't want to change it. I suggest either discussing this feature in the commonmark discussion forums: https://talk.commonmark.org/  Or writing an extension that doesn't intefere with the core library. I'm closing this PR for now.